catalyst-gateway/bin/src/service/api/documents/mod.rs

```diff
@@ -71,17 +76,17 @@ impl DocumentApi {
     )]
     async fn put_document(
         &self, /// The document to PUT
-        document: Cbor<Body>,
+        cbor: Cbor,
         /// Authorization required.
-        _auth: CatalystRBACSecurityScheme,
+        _auth: NoneOrRBAC,
     ) -> put_document::AllResponses {
-        match document.0.into_bytes_limit(MAXIMUM_DOCUMENT_SIZE).await {
-            Ok(document) => put_document::endpoint(document).await,
+        match cbor.into_bytes_with_limit(MAXIMUM_DOCUMENT_SIZE).await {
+            Ok(doc_bytes) => put_document::endpoint(doc_bytes).await,
             Err(ReadBodyError::PayloadTooLarge) => put_document::Responses::PayloadTooLarge.into(),
-            Err(_err) => {
-                put_document::Responses::BadRequest(Json(PutDocumentBadRequest::new(
-                    "Failed to read document from the request",
-                )))
+            Err(e) => {
+                put_document::Responses::BadRequest(Json(PutDocumentBadRequest::new(format!(
```

Comment:
Don't put internal errors into 400 responses by serializing Err.
Deserializing errors is a hack vector, whether you think its hackable or not its very bad practice from a security perspective.